### DISCUSSION TITLE
Same filters and diff extensions but diff results, why?

### DISCUSSION THREAD

**Discussion by kodiakhub** [uBlockOrigin/uAssets#24340] [Q&A]
There is no issue if uBO filters are enabled on the uBO side, but why is this so? Thanks in advance.

Tested site: `mackolik.com/canli-sonuclar`

<details><summary>Screenshot</summary>
<p>

| AdGuard | uBO |
| ------------- | ------------- |
| ![image](https://github.com/uBlockOrigin/uAssets/assets/56846906/1b5d79cf-5bd0-4533-ad06-8ab776ceada9) | ![image](https://github.com/uBlockOrigin/uAssets/assets/56846906/4aedb505-3690-4c7d-9b5f-925f663173ff) |
| ![image](https://github.com/uBlockOrigin/uAssets/assets/56846906/0b3fb645-e87e-4329-9446-ba8ee7827e98) | ![image](https://github.com/uBlockOrigin/uAssets/assets/56846906/28a90be0-ad35-44e1-bb9e-440cdc1cd305) |

</p>
</details>

Edit: I forgot to disable a filter you see in the picture on the uBO side. I tested again and same result.

<details><summary>Screenshot</summary>
<p>

![image](https://github.com/uBlockOrigin/uAssets/assets/56846906/4942c7c9-1f0d-42f3-afbf-a6da68d3f635)

</p>
</details>

Update: If i disable EasyList filter too, no issue. But in AdGuard extension, EasyList filter is included in AdGuard Base filter list, how does this happen? Am I missing something? 😄

**Comment by stephenhawk8054** [comment]
I can't reproduce

![image](https://github.com/uBlockOrigin/uAssets/assets/66517106/e7b38221-042f-45f8-9590-df6a25071ade)

Test again in a new browser profile with only uBO and its default settings.

**Reply by kodiakhub** [reply]
@stephenhawk8054 I've already disabled uBO filters on purpose, that's what my question was about. 😄

**Reply by stephenhawk8054** [reply]
Sorry. I don't understand what you mean by `disabled uBO filters`.

Does the issue occur on a new profile with only uBO and its default settings?

**Reply by kodiakhub** [reply]
> Sorry. I don't understand what you mean by `disabled uBO filters`.
> 
> Does the issue occur on a new profile with only uBO and its default settings?

Ofc does not occur with default settings. It was a deliberate attempt and I wondered why this was happening?

**Reply by kodiakhub** [reply]
Sorry, I am writing with translation. I'm sorry if you misunderstand. 😄 

I just tried the same AdGuard filters in uBO and the results were different.

**Reply by stephenhawk8054** [reply]
Ok, I understand what you mean now.

In `uBlock filters – Ads`, there's this filter

```adb
||fundingchoicesmessages.google.com^$3p
```

which prevents the anti-adblock popup. You can mimic the popup by disabling it when adding this filter

```adb
||fundingchoicesmessages.google.com^$3p,badfilter
```

**Reply by kodiakhub** [reply]
I think I'm confused. Because I didn't get the answer I expected. It seems like it would be clearer if I showed it again with a video:

<details><summary>Video</summary>
<p>

https://github.com/uBlockOrigin/uAssets/assets/56846906/73a4fba4-1cb5-46e8-890f-a23eee5c75c4

</p>
</details>

I was just wondering why it didn't give the same result without the uBO filters.

**Reply by garry-ut99** [reply]
Given `AdGuard – Ads` (base filter) filter list is sufficient to hide the popup in AdGuard extension, then it seems what OP is aksing, is why the popup still appears in uBO extension with the same `AdGuard – Ads` filter list enabled.

Perhaps some filters in `AdGuard – Ads` responsible for hiding the popup work in AdGuard but not in uBlock, might be a bug in uBO, or filters incompatibility between uBO and AdGuard, or the filters which hides the popup are not present in uBO version of `AdGuard – Ads`.

 It needs further investigation to find the cause, a good starting point is to find which filter in `AdGuard – Ads` (base filter) is responsible for hiding the popup in AdGuard, then see whether the same filter exist in uBO version of `AdGuard – Ads`.

**Reply by kodiakhub** [reply]
@garry-ut99 Exactly as you said, you explained it perfectly. Thank you. 😄

**Reply by Yuki2718** [reply]
I didn't test the case but as said, whether blocking `fundingchoicesmessages.google.com` or not makes the difference. With it blocked, these are what are necessary to defuse GFC anti-adblock aka ad-blocking recovery message:
```
#@#[class^="div-gpt-ad"]
#@#[id^="div-gpt-ad"]
#@#div[id^="div-gpt-"]
##[class^="div-gpt-ad"]:not([style^="width: 1px; height: 1px; position: absolute; left: -10000px; top: -"])
##[id^="div-gpt-ad"]:not([style^="width: 1px; height: 1px; position: absolute; left: -10000px; top: -"])
##div[id^="div-gpt-"]:not([style^="width: 1px; height: 1px; position: absolute; left: -10000px; top: -"])
```

with it unblocked, this one
```
#$#div[class*=" "][style^="width: 1px; height: 1px; position: absolute; left: -10000px; top: -"] { display: block !important; }
```

Each of filter list is built upon specific assumption of each blocker and such differences appear also about e.g. `googletagmanager.com` which uBO assumes it's blocked while AG not unless one enabled Tracking Protection filter.

**Reply by kodiakhub** [reply]
Just in case, I hope the Filter logs from two separate extensions can help.

<details><summary>AdGuard Filtering log</summary>

![AdGuard](https://github.com/uBlockOrigin/uAssets/assets/56846906/72245acc-c9ca-42d4-9226-d369ee2ce99d)

</details>

<details><summary>uBO Logger output</summary>

```
+1
www.mackolik.com
1
get
image
https://www.mackolik.com/favicons/mackolik/android-icon-192x192.c057bdbd3f.png
```
```
+1
.ads.css
--
www.mackolik.com
3
get
css
https://cdn.jsdelivr.net/npm/videojs-contrib-ads@6.7.0/dist/videojs.ads.css
```
```
+1
www.mackolik.com
3
post
ping
https://ym-tack.b-cdn.net/load_playlist_success?sellbe;MKL;https://cdn.membrana.media/video/mkl/desktop/videoSources.json;13bd8d29-6f05-4341-9b4b-371814164503
```
```
+1
www.mackolik.com
3
post
xhr
https://kinesis.us-east-2.amazonaws.com/
```
```
+1
www.mackolik.com
3
post
xhr
https://fundingchoicesmessages.google.com/el/AGSKWxXGjfDmjumbTnhFF8DmPocNMRsYuIb8cbEc15KEomgu8I7r5fPWQPrBCpZhwWl90Gjmm0J0YQWtIvtnCEUh_mJ_avum6cXDe37wLlBfEbFqsKFBc0Cw44kZbwcgssAkUIqdxykhyA==
```
```
+1
||imasdk.googleapis.com^
--
www.mackolik.com
3
get
script
https://imasdk.googleapis.com/js/sdkloader/ima3.js
```
```
+1
www.mackolik.com
3
get
media
https://video-content.b-cdn.net/common/1-second-of-silence.mp3
```
```
+1
||adpartner.pro^
--
www.mackolik.com
3
get
image
https://a4p.adpartner.pro/ssp/match?redirect=https://bidder.smartytouch.co/icm?uid=%7Buser_id%7D
```
```
+1
||adpartner.pro^
--
www.mackolik.com
3
get
image
https://a4p.adpartner.pro/ssp/match?redirect=https%3A%2F%2Fpixel.onaudience.com%2F%3Fpartner%3D283%26mapped%3D%7Buser_id%7D%26gdpr%3D0
```
```
+1
||adpartner.pro^
--
www.mackolik.com
3
get
image
https://a4p.adpartner.pro/ssp/match?redirect=https://server.smartytech.io/icm.gif?uid=%7Buser_id%7D
```
```
+1
www.mackolik.com
3
get
xhr
https://cdn.membrana.media/video/mkl/desktop/videoSources.json
```
```
+1
www.mackolik.com
3
post
ping
https://ym-tack.b-cdn.net/load_playlist_attempt?sellbe;MKL;https://cdn.membrana.media/video/mkl/desktop/videoSources.json;13bd8d29-6f05-4341-9b4b-371814164503
```
```
+1
www.mackolik.com
3
post
xhr
https://kinesis.us-east-2.amazonaws.com/
```
```
+1
www.mackolik.com
3
get
xhr
https://cdn.membrana.media/geolocation.json
```
```
+1
www.mackolik.com
3
post
xhr
https://kinesis.us-east-2.amazonaws.com/
```
```
+1
www.mackolik.com
3
get
frame
https://fundingchoicesmessages.google.com/s/whitelist?hl=tr
```
```
+1
www.mackolik.com
3
post
xhr
https://fundingchoicesmessages.google.com/el/AGSKWxV1T_hektPNl_Mlq8jVsnsdAoyanXfUPVqTmJQv77KwwhPltlGViWvQMCF-9YNh1BaDRjefvnoYAGRjCFPZoV2I7NkHnBo-P6eSaTOpOa1xnage4Agg_AnXHjig41Tnw_30VLbiUg==
```
```
+1
www.mackolik.com
3
get
script
https://fundingchoicesmessages.google.com/f/AGSKWxUlT562ekrdOf46UqZPPG4VZApm5FJOuqJXSC4B8x2A2JD1oBGhY4yNJNShS8gbyFDn_sanwxpr_zMgvfmZZ18W4UY2VPmAgBITSyry8jolo-IvnMOkVIiuA6o5Q8BGajmtdAzW8w==?fccs=W1siQUtzUm9sODdJT2ZUTWpEYnRGZy1LU0lLS3N0azV6UDhTblY5TzNmckRKWS1TLU1QRlhqUU9pVWNDOUVfbXJkN0dSTGRtS2ZZbnlhb0t5c3NvalJGZThPZW5RRGtIOTRuYTg4RmF3OXhCVHlUalUzMkJPRXNJeHdVYWVkQWZJMW12NnVicVVVNXMzMUZfbmdmS3JhMTFqQnBEbm9DV2dKTjhRPT0iXSxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsWzE3MTk5MTcyMDksMTIwMDAwMDAwXSxudWxsLG51bGwsbnVsbCxbbnVsbCxbNyw2LDVdLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLDJdLCJodHRwczovL3d3dy5tYWNrb2xpay5jb20vY2FubGktc29udWNsYXIiLG51bGwsW1s4LCJXcXJCallwa1JaUSJdLFs5LCJlbiJdLFsxNiwiWzEsMSwxXSJdLFsyMiwiZmFsc2UiXSxbMjAsIltudWxsLG51bGwsWzMxMDg0MTg4XSxudWxsLDE2XSJdLFsxOSwiMiJdLFsxNywiWzBdIl1dXQ
```
```
+0
www.mackolik.com
3
get
script
https://fundingchoicesmessages.google.com/f/AGSKWxX7VFnwyZ6rp6zCTMSbRi2RoPJx1irmE2SkBQXtIYJrzxdYZXkrVN4j_7ZAAgwgq6mBaKysPWiCS0O4miy5J58fb9DIynhBMVtLdzgUPzF0jAcA0yHbNlPpBgr7SxUhwHk5rUgAcQ==?fccs=W1siQUtzUm9sODdJT2ZUTWpEYnRGZy1LU0lLS3N0azV6UDhTblY5TzNmckRKWS1TLU1QRlhqUU9pVWNDOUVfbXJkN0dSTGRtS2ZZbnlhb0t5c3NvalJGZThPZW5RRGtIOTRuYTg4RmF3OXhCVHlUalUzMkJPRXNJeHdVYWVkQWZJMW12NnVicVVVNXMzMUZfbmdmS3JhMTFqQnBEbm9DV2dKTjhRPT0iXSxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsWzE3MTk5MTcyMDgsOTkyMDAwMDAwXSxudWxsLG51bGwsbnVsbCxbbnVsbCxbNyw2XSxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCwyXSwiaHR0cHM6Ly93d3cubWFja29saWsuY29tL2NhbmxpLXNvbnVjbGFyIixudWxsLFtbOCwiV3FyQmpZcGtSWlEiXSxbOSwiZW4iXSxbMTYsIlsxLDEsMV0iXSxbMjIsImZhbHNlIl0sWzIwLCJbbnVsbCxudWxsLFszMTA4NDE4OF0sbnVsbCwxNl0iXSxbMTksIjIiXSxbMTcsIlswXSJdXV0
```
```
+0
www.mackolik.com
3
post
xhr
https://fundingchoicesmessages.google.com/el/AGSKWxV1T_hektPNl_Mlq8jVsnsdAoyanXfUPVqTmJQv77KwwhPltlGViWvQMCF-9YNh1BaDRjefvnoYAGRjCFPZoV2I7NkHnBo-P6eSaTOpOa1xnage4Agg_AnXHjig41Tnw_30VLbiUg==
```
```
+0
##.shadvertisment
www.mackolik.com
get
dom
https://www.mackolik.com/canli-sonuclar
```
```
+0
##.ad-bg
www.mackolik.com
get
dom
https://www.mackolik.com/canli-sonuclar
```
```
+0
www.mackolik.com
1
get
image
https://www.mackolik.com/favicons/mackolik/favicon-32x32.701a97cccb.png
```
```
+0
www.mackolik.com
1
get
image
https://www.mackolik.com/favicons/mackolik/favicon.e60ce9af22.ico
```
```
+0
www.mackolik.com
1
get
other
https://www.mackolik.com/favicons/mackolik/manifest.baf78f2af3.json
```
```
+0
www.mackolik.com
3
get
script
https://fundingchoicesmessages.google.com/f/AGSKWxUw80fCvc6eAxfY53v82BIjtTd0UjCcimXPmQpGM_n5ZjtZpi_JvGW8u4TElJTY-E4WipciVFZFSH3ZHyFfUWpX_NENMBt7u3PYAU0V8qWq_ruFGGRIMzWWXMFvWlEZXvMoiBedGA==?fccs=W1siQUtzUm9sODdJT2ZUTWpEYnRGZy1LU0lLS3N0azV6UDhTblY5TzNmckRKWS1TLU1QRlhqUU9pVWNDOUVfbXJkN0dSTGRtS2ZZbnlhb0t5c3NvalJGZThPZW5RRGtIOTRuYTg4RmF3OXhCVHlUalUzMkJPRXNJeHdVYWVkQWZJMW12NnVicVVVNXMzMUZfbmdmS3JhMTFqQnBEbm9DV2dKTjhRPT0iXSxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsWzE3MTk5MTcyMDgsNDIwMDAwMDAwXSxudWxsLG51bGwsbnVsbCxbbnVsbCxbN11dLCJodHRwczovL3d3dy5tYWNrb2xpay5jb20vY2FubGktc29udWNsYXIiLG51bGwsW1s4LCJXcXJCallwa1JaUSJdLFs5LCJlbiJdLFsxNiwiWzEsMSwxXSJdLFsyMiwiZmFsc2UiXSxbMjAsIltudWxsLG51bGwsWzMxMDg0MTg4XSxudWxsLDE2XSJdLFsxOSwiMiJdLFsxNywiWzBdIl1dXQ
```
```
+0
www.mackolik.com
3
get
xhr
https://api.mackolikfeeds.com/socket/?scheme=https&application=mackolik.com
```
```
+0
www.mackolik.com
1
get
xhr
https://www.mackolik.com/ajax/middleware/token
```
```
+0
www.mackolik.com
1
get
xhr
https://www.mackolik.com/perform/p0/ajax/components/competition/livescores/json?sports[]=Soccer&sports[]=Basketball&sports[]=Tennis&matchDate=2024-07-02
```
```
+0
##div[id^="div-gpt-"]
www.mackolik.com
get
dom
https://www.mackolik.com/canli-sonuclar
```
```
+0
##div[class^="widget-social"]
www.mackolik.com
get
dom
https://www.mackolik.com/canli-sonuclar
```
```
+0
##[id^="div-gpt-ad"]
www.mackolik.com
get
dom
https://www.mackolik.com/canli-sonuclar
```
```
+0
##.widget-gpt-ads--topbanner
www.mackolik.com
get
dom
https://www.mackolik.com/canli-sonuclar
```
```
+0
##.widget-gdpr-banner
www.mackolik.com
get
dom
https://www.mackolik.com/canli-sonuclar
```
```
+0
##.sidebar > .mpu--2
www.mackolik.com
get
dom
https://www.mackolik.com/canli-sonuclar
```
```
+0
##.mpu
www.mackolik.com
get
dom
https://www.mackolik.com/canli-sonuclar
```
```
+0
###taboola-below-article-thumbnails
www.mackolik.com
get
dom
https://www.mackolik.com/canli-sonuclar
```
```
+0
||hit.gemius.pl^
--
www.mackolik.com
3
get
script
https://gatr.hit.gemius.pl/xgemius.js
```
```
+0
www.mackolik.com
1
get
script
https://www.mackolik.com/js/livescoreDataWorker.75977fff6d.worker.js
```
```
+0
||googletagmanager.com/gtag/$3p
--
www.mackolik.com
3
get
script
https://www.googletagmanager.com/gtag/js?id=G-L6S4CM0W94&l=ymDataLayer
```
```
+0
||adpartner.pro^
--
www.mackolik.com
3
get
image
https://a4p.adpartner.pro/ssp/match?redirect=https://bidder.smartytouch.co/icm?uid=%7Buser_id%7D
```
```
+0
||adpartner.pro^
--
www.mackolik.com
3
get
image
https://a4p.adpartner.pro/ssp/match?redirect=https%3A%2F%2Fpixel.onaudience.com%2F%3Fpartner%3D283%26mapped%3D%7Buser_id%7D%26gdpr%3D0
```
```
+0
||adpartner.pro^
--
www.mackolik.com
3
get
image
https://a4p.adpartner.pro/ssp/match?redirect=https://server.smartytech.io/icm.gif?uid=%7Buser_id%7D
```
```
+0
www.mackolik.com
3
get
xhr
https://cdn.jsdelivr.net/gh/prebid/currency-file@1/latest.json?date=20240702
```
```
+0
www.mackolik.com
3
post
xhr
https://kinesis.us-east-2.amazonaws.com/
```
```
+0
www.mackolik.com
3
get
xhr
https://cdn.membrana.media/geolocation.json
```
```
+0
||mc.yandex.ru^
--
www.mackolik.com
3
get
script
https://mc.yandex.ru/metrika/tag.js
```
```
+0
www.mackolik.com
3
get
script
https://fundingchoicesmessages.google.com/f/AGSKWxVp7LOl-jg6iR6ek9kHeuwo0xwGrkQLMhdUYuiGScNesdA5QabkA_CQZnOct9Ag9S5qd_m9HuYimGF9eDb1Bsg=
```
```
+0
||taboola.com^$3p
--
www.mackolik.com
3
get
script
https://cdn.taboola.com/libtrc/makolik/loader.js
```
```
+0
||googletagmanager.com/*=gtm-
--
www.mackolik.com
3
get
script
https://www.googletagmanager.com/gtm.js?id=GTM-MF7ZLM4
```
```
+0
||pagead2.googlesyndication.com^
--
www.mackolik.com
3
get
script
https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js
```
```
+0
/prebid8.$script
--
www.mackolik.com
1
get
script
https://www.mackolik.com/js/prebid8.40.0.js
```
```
+0
googletagservices_gpt.js
<<
www.mackolik.com
3
get
script
https://securepubads.g.doubleclick.net/tag/js/gpt.js
```
```
+0
||securepubads.g.doubleclick.net/tag/js/gpt.js$script,redirect-rule=googletagservices_gpt.js,from=~eksisozluk.com|~golf.com|~heycar.co.uk|~inmuebles24.com|~inside.com.tw|~investing.com|~olx.com.eg|~today.line.me
--
www.mackolik.com
3
get
script
https://securepubads.g.doubleclick.net/tag/js/gpt.js
```
```
+0
||doubleclick.net^
--
www.mackolik.com
3
get
script
https://securepubads.g.doubleclick.net/tag/js/gpt.js
```
```
+0
www.mackolik.com
1
get
doc
https://www.mackolik.com/canli-sonuclar
```
```
+0
https://www.mackolik.com/canli-sonuclar
```
</details>

**Reply by garry-ut99** [reply]
Regardless of what is the reason of the current specific issue,  

speaking generally: 100% compatibility all the time between uBO and AG extensions, even when using the same filter list, is not guaranteed, because some filters might be just not 100% compatible, like it's not guaranteed that the same website will work/look 100% same in different browsers (for example: Firefox vs Chrome).

**Reply by kodiakhub** [reply]
> Regardless of what is the reason of the current specific issue,
> 
> speaking generally: 100% compatibility all the time between uBO and AG extensions, even when using the same filter list, is not guaranteed, because some filters might be just not 100% compatible, like it's not guaranteed that the same website will work/look 100% same in different browsers (for example: Firefox vs Chrome).

Your comment is understandable enough. I think there is no need to go deep on this topic then. 🙂

**Reply by Yuki2718** [reply]
Tested and it's certainly GFC. If you disabled uBf and enabled AG Base, you need `||fundingchoicesmessages.google.com^$3p` and the aformentioned filters. Since the bait elements bail out too soon, they may not appear in logs. Or if you prefer AG's way of not blocking `fundingchoicesmessages.google.com`, `*##div[class*=" "][style^="width: 1px; height: 1px; position: absolute; left: -10000px; top: -"]:style(display: block !important;)`

**Reply by garry-ut99** [reply]
> Yuki2718 : `#$#div[class*=" "][style^="width: 1px; height: 1px; position: absolute; left: -10000px; top: -"] { display: block !important; `

That filter is already present in `AdGuard – Ads` in both AdGuard and uBlock, and should work in both, because it's compatible with uBO too, but for some reason it works only in AdGuard, but not in uBO, and this is what is the reason of the current reported issue. 

Prepending the filter by `*` makes it working in uBO in the current issue, because it makes the filter to be injected earlier (unconditionally): `*#$#div[class*=" "][style^="width: 1px; height: 1px; position: absolute; left: -10000px; top: -"] { display: block !important; }`.

The reported issue might be caused by internal differences between AdGuard's and uBlock's cosmetic filtering engines: for ex. how soon they inject cosmetic filters, or by some other factors.

<hr>

By the way, alternative ways to fix the AA are:
-  to add: `@@||mackolik.com^$ghide` in uBO, like mentioned in:
> ! NOTE: contributor.google.com
> ! generic fix for e.g. class="subadimg footerFullAd pdpads_desktop adkit-advert adline"
> ! IMPORTANT: Add generichide exception for content blockers to `content-blocker.txt` in the same section
> `#$#div[class*=" "][style^="width: 1px; height: 1px; position: absolute; left: -10000px; top: -"] { display: block !important; }`

- or the way it's fixed in `uBlock filters – Ads`, by several cosmetic exceptions, instead of `ghide`

But these are just alternatives, the point of the case reported by OP is why the anti-adblock message is not hidden by filtes which are already present on `AdGuard – Ads` list, without adding or modifying filters, and it has been explained in the upper part of my comment.

**Reply by Yuki2718** [reply]
It's not "earlier", uBO doesn't support generic css filter without `*`. ~~Maybe will work if you set `allowGenericProceduralFilters` to `true`~~ <= apparently not, as it's not a procedural cosmetic.

**Reply by garry-ut99** [reply]
Seems so:

From Wiki:
> https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#subjectstylearg: As with the other new cosmetic filtering selectors, `:style` can be used only for _specific_ cosmetic filters. A hostname or entity must get specified for the filter.

Also from Logger:

> Invalid generic cosmetic filter in adguard-generic: `##div[class*=" "][style^="width: 1px; height: 1px; position: absolute; left: -10000px; top: -"] { display: block !important; }`

Mystery solved.

By the way, "My filters" linter doesn't warn, I'm not sure whether it should or not.

**Comment by kodiakhub** [comment]
@stephenhawk8054 You previously wrote this clarification regarding Facebook issues, see: https://github.com/uBlockOrigin/uAssets/issues/3367#issue-355645933

Currently, Facebook ads and sponsored content can be blocked using the AdGuard extension and AdGuard filters. Although some issues arise, they are eventually resolved, making the overall situation positive. However, on the current uBO side, Facebook is still displaying sponsored ads.

Given this, I’m wondering whether the Facebook-specific rules in the AdGuard Base filter list can be adapted for compatibility with uBlock Origin (uBO) under these conditions. 🤔

I ask because there are noticeable differences in functionality and performance between the AdGuard and uBO extensions. I have personally observed these differences while testing various websites (e.g., race conditions, slow responses, and memory leaks).

Additionally, AdGuard filters are frequently updated regardless of the workload, whereas I believe this may not be feasible on the uBO side due to certain constraints.

**Reply by stephenhawk8054** [reply]
AG is using raw JS for Facebook, so it's not available in AG list on uBO.

**Reply by stephenhawk8054** [reply]
Again, please understand this:

The issue is not about SOLUTION. Everything has a solution. I can find a solution for Facebook, no problem.

The issue is about whatever solution we put into uBO, facebook counters that. It doesn't matter if the solution comes from AG or any other extensions. Remember, we are volunteers, I don't have time to monitor facebook 24/7 every single second of the day. If there's ads issue, please report it. If there's no one to report it, we cannot solve it.

The comment in the facebook thread is about warning that the solutions won't be stable or breakage-free. That's it.

**Reply by kodiakhub** [reply]
> Remember, we are volunteers, I don't have time to monitor facebook 24/7 every single second of the day. If there's ads issue, please report it. If there's no one to report it, we cannot solve it.

Okay, I understand you, but what I mean is that I was expecting AdGuard filters to function with full compatibility in the uBO extension. So in this case, the responsibility would lie with AdGuard rather than uBO. This is what I mean.

So, the "raw JS" thing is not an easy process for compatibility, or whatever is preventing it, it's not an easy process, right? 🤔

**Reply by stephenhawk8054** [reply]
Raw JS cannot be updated via filter lists, because it would violate extension stores' no remote code execution policy. They have to be updated with the whole extension. Previously AG can update raw JS on chrome store only due to the store doesn't care about reviewing, but now with MV3, google doesn't allow it any more either. Firefox store is completely prohibited.

**Reply by gorhill** [reply]
I have been thinking of supporting raw `#%#` JS by converting these to formal scriptlets with build-time generated hash-based names. However I am not sure the amount of work is worth the results, and keeping in mind working on this would prevent me from working on maybe more useful other stuff (for example the new jsonpath syntax).

f list maintainers think it's would be a top improvement to have versus other things to do, I am willing to work on this. Also, this would have to be filters available only from trusted sources, so AG wouldn't qualify since only uBO's own lists are to be trusted.

**Reply by stephenhawk8054** [reply]
> build-time generated hash-based names

I actually had one concern with raw JS when you discussed about it, which is what is the best way to make exceptions to the raw JS filter via quick update filter lists, since the raw JS is integrated to the extension (I assume that those would still have to be updated via extensions, is it correct) ?

The hash-based names might make sense I think if I understand correctly. So something like this:

```adb
example.com#@#+js(somehash)
```

?

**Reply by gorhill** [reply]
Yes, assembled at build time, thus updated only via new published versions. Also yes, generated names would be used for exceptions. At runtime, uBO would use the same algorithm to derive name, then would use this to reference the scriptlet, hence can be used for exceptions or even to reuse the raw scriptlet for other sites if needed.

I gather sometimes no scriptlet fulfill the job while a small bit of raw JS would work. Downside is this requires an extension update, so uBOL would benefit most from this. Anyways, I need to know whether this is something that would really be useful to have.

**Reply by stephenhawk8054** [reply]
Yeah, it's useful for some `rpnt` filters. But I think jsonpath is still more needed now, since raw JS is not too effective for sites that counter regularly, even for uBOL.
